[no-unused-vars] uses non-existent deep import from eslint@5

<!--
If you have a problem with a specific rule, please begin your issue title with [rulename] to make it easier to search for.
I.e. "[no-unused-vars] False positive when fooing the bar"

Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
Are you opening an issue because the rule you're trying to use is not found?
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨
1) Check the releases log: https://github.com/typescript-eslint/typescript-eslint/releases
    -  If the rule isn't listed there, then chances are it hasn't been released to the main npm tag yet.
2) Try installing the `canary` tag: `npm i @typescript-eslint/eslint-plugin@canary`.
    - The canary tag is built for every commit to master, so it contains the bleeding edge build.
3) If ESLint still can't find the rule, then consider reporting an issue.
-->

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.

Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

Make sure to install `eslint@5`, which is supported according to peer dependency: https://github.com/typescript-eslint/typescript-eslint/blob/717e718e91df2165422228c02dfa248cf55f65a1/packages/eslint-plugin/package.json#L64

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

```TS
// empty
```

<!--
Also include your tsconfig, if you're using type-aware linting
-->

**Expected Result**

The linter should exit successfully

**Actual Result**

It fails due to this import, which doesn't exist for eslint@5: 

https://github.com/typescript-eslint/typescript-eslint/blob/717e718e91df2165422228c02dfa248cf55f65a1/packages/eslint-plugin/src/rules/no-unused-vars.ts#L7

**Additional Info**

<!--
Did eslint throw an exception?

Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug
-->

<details>
<summary>Details</summary>

```
eslint src/index.ts --debug
  eslint:cli CLI args: [ 'src/index.ts', '--debug' ] +0ms
  eslint:cli Running on files +4ms
  eslint:glob-utils Creating list of files to process. +0ms
  eslint:ignored-paths baseDir = "/Users/simen/repos/eslint-plugin-jest" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/node_modules/*"
  eslint:ignored-paths   cooked   = "/node_modules/*" +1ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/bower_components/*"
  eslint:ignored-paths   cooked   = "/bower_components/*" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = ".*"
  eslint:ignored-paths   cooked   = ".*" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "!../"
  eslint:ignored-paths   cooked   = "!../" +0ms
  eslint:ignored-paths Looking for ignore file in /Users/simen/repos/eslint-plugin-jest +0ms
  eslint:ignored-paths Loaded ignore file /Users/simen/repos/eslint-plugin-jest/.eslintignore +1ms
  eslint:ignored-paths Adding /Users/simen/repos/eslint-plugin-jest/.eslintignore +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "coverage/"
  eslint:ignored-paths   cooked   = "coverage/" +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "lib/"
  eslint:ignored-paths   cooked   = "lib/" +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "!.eslintrc.js"
  eslint:ignored-paths   cooked   = "!.eslintrc.js" +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "coverage/"
  eslint:ignored-paths   cooked   = "coverage/" +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "lib/"
  eslint:ignored-paths   cooked   = "lib/" +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "!.eslintrc.js"
  eslint:ignored-paths   cooked   = "!.eslintrc.js" +0ms
  eslint:ignored-paths contains: +2ms
  eslint:ignored-paths   target = "/Users/simen/repos/eslint-plugin-jest/src/index.ts" +0ms
  eslint:ignored-paths   result = false +0ms
  eslint:ignored-paths contains: +0ms
  eslint:ignored-paths   target = "/Users/simen/repos/eslint-plugin-jest/src/index.ts" +0ms
  eslint:ignored-paths   result = false +0ms
  eslint:cli-engine Processing /Users/simen/repos/eslint-plugin-jest/src/index.ts +0ms
  eslint:cli-engine Linting /Users/simen/repos/eslint-plugin-jest/src/index.ts +1ms
  eslint:config Constructing config file hierarchy for /Users/simen/repos/eslint-plugin-jest/src +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /Users/simen/repos/eslint-plugin-jest/.eslintrc.js +1ms
  eslint:config-file Loading JS config file: /Users/simen/repos/eslint-plugin-jest/.eslintrc.js +0ms
  eslint:plugins Loaded plugin eslint-config (eslint-plugin-eslint-config@1.0.4) (from /Users/simen/repos/eslint-plugin-jest/node_modules/eslint-plugin-eslint-config/lib/index.js) +0ms
  eslint:plugins Loaded plugin eslint-plugin (eslint-plugin-eslint-plugin@2.3.0) (from /Users/simen/repos/eslint-plugin-jest/node_modules/eslint-plugin-eslint-plugin/lib/index.js) +8ms
  eslint:plugins Loaded plugin eslint-comments (eslint-plugin-eslint-comments@3.2.0) (from /Users/simen/repos/eslint-plugin-jest/node_modules/eslint-plugin-eslint-comments/index.js) +8ms
  eslint:plugins Loaded plugin node (eslint-plugin-node@11.1.0) (from /Users/simen/repos/eslint-plugin-jest/node_modules/eslint-plugin-node/lib/index.js) +67ms
  eslint:plugins Loaded plugin prettier (eslint-plugin-prettier@3.3.0) (from /Users/simen/repos/eslint-plugin-jest/node_modules/eslint-plugin-prettier/eslint-plugin-prettier.js) +2ms
  eslint:plugins Loaded plugin import (eslint-plugin-import@2.22.1) (from /Users/simen/repos/eslint-plugin-jest/node_modules/eslint-plugin-import/lib/index.js) +78ms
Error: Cannot find module 'eslint/lib/rules/utils/ast-utils'
Require stack:
- /Users/simen/repos/eslint-plugin-jest/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-unused-vars.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/eslint/lib/config/plugins.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/eslint/lib/config.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/eslint/lib/cli-engine.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/eslint/lib/cli.js
- /Users/simen/repos/eslint-plugin-jest/node_modules/eslint/bin/eslint.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/simen/repos/eslint-plugin-jest/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-unused-vars.js:24:21)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
```

</details>

The regression was introduced in #2768.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.10.0` |
| `@typescript-eslint/parser`        | `4.10.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `14.15.0` |

--- a/packages/eslint-plugin/package.json
+++ b/packages/eslint-plugin/package.json
@@ -46,6 +46,7 @@
     "@typescript-eslint/scope-manager": "4.12.0",
     "debug": "^4.1.1",
     "functional-red-black-tree": "^1.0.1",
+    "lodash": "^4.17.15",
     "regexpp": "^3.0.0",
     "semver": "^7.3.2",
     "tsutils": "^3.17.1"

--- a/packages/eslint-plugin/src/rules/no-redeclare.ts
+++ b/packages/eslint-plugin/src/rules/no-redeclare.ts
@@ -13,43 +13,6 @@ type Options = [
   },
 ];
 
-// https://github.com/lodash/lodash/blob/86a852fe763935bb64c12589df5391fd7d3bb14d/escapeRegExp.js
-const reRegExpChar = /[\\^$.*+?()[\]{}|]/g;
-const reHasRegExpChar = RegExp(reRegExpChar.source);
-function escapeRegExp(str: string): string {
-  return str && reHasRegExpChar.test(str)
-    ? str.replace(reRegExpChar, '\\$&')
-    : str || '';
-}
-
-function getNameLocationInGlobalDirectiveComment(
-  sourceCode: TSESLint.SourceCode,
-  comment: TSESTree.Comment,
-  name: string,
-): TSESTree.SourceLocation {
-  const namePattern = new RegExp(
-    `[\\s,]${escapeRegExp(name)}(?:$|[\\s,:])`,
-    'gu',
-  );
-
-  // To ignore the first text "global".
-  namePattern.lastIndex = comment.value.indexOf('global') + 6;
-
-  // Search a given variable name.
-  const match = namePattern.exec(comment.value);
-
-  // Convert the index to loc.
-  const start = sourceCode.getLocFromIndex(
-    comment.range[0] + '/*'.length + (match ? match.index + 1 : 0),
-  );
-  const end = {
-    line: start.line,
-    column: start.column + (match ? name.length : 1),
-  };
-
-  return { start, end };
-}
-
 export default util.createRule<Options, MessageIds>({
   name: 'no-redeclare',
   meta: {
@@ -129,7 +92,7 @@ export default util.createRule<Options, MessageIds>({
           yield {
             type: 'comment',
             node: comment,
-            loc: getNameLocationInGlobalDirectiveComment(
+            loc: util.getNameLocationInGlobalDirectiveComment(
               sourceCode,
               comment,
               variable.name,

--- a/packages/eslint-plugin/src/rules/no-unused-vars.ts
+++ b/packages/eslint-plugin/src/rules/no-unused-vars.ts
@@ -4,7 +4,6 @@ import {
   TSESTree,
 } from '@typescript-eslint/experimental-utils';
 import { PatternVisitor } from '@typescript-eslint/scope-manager';
-import { getNameLocationInGlobalDirectiveComment } from 'eslint/lib/rules/utils/ast-utils';
 import * as util from '../util';
 
 export type MessageIds = 'unusedVar';
@@ -388,7 +387,7 @@ export default util.createRule<Options, MessageIds>({
 
             context.report({
               node: programNode,
-              loc: getNameLocationInGlobalDirectiveComment(
+              loc: util.getNameLocationInGlobalDirectiveComment(
                 sourceCode,
                 directiveComment,
                 unusedVar.name,

--- a/packages/eslint-plugin/src/util/astUtils.ts
+++ b/packages/eslint-plugin/src/util/astUtils.ts
@@ -1,2 +1,43 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+import escapeRegExp from 'lodash/escapeRegExp';
+
 // deeply re-export, for convenience
 export * from '@typescript-eslint/experimental-utils/dist/ast-utils';
+
+// The following is copied from `eslint`'s source code since it doesn't exist in eslint@5.
+// https://github.com/eslint/eslint/blob/145aec1ab9052fbca96a44d04927c595951b1536/lib/rules/utils/ast-utils.js#L1751-L1779
+// Could be export { getNameLocationInGlobalDirectiveComment } from 'eslint/lib/rules/utils/ast-utils'
+/**
+ * Get the `loc` object of a given name in a `/*globals` directive comment.
+ * @param {SourceCode} sourceCode The source code to convert index to loc.
+ * @param {Comment} comment The `/*globals` directive comment which include the name.
+ * @param {string} name The name to find.
+ * @returns {SourceLocation} The `loc` object.
+ */
+export function getNameLocationInGlobalDirectiveComment(
+  sourceCode: TSESLint.SourceCode,
+  comment: TSESTree.Comment,
+  name: string,
+): TSESTree.SourceLocation {
+  const namePattern = new RegExp(
+    `[\\s,]${escapeRegExp(name)}(?:$|[\\s,:])`,
+    'gu',
+  );
+
+  // To ignore the first text "global".
+  namePattern.lastIndex = comment.value.indexOf('global') + 6;
+
+  // Search a given variable name.
+  const match = namePattern.exec(comment.value);
+
+  // Convert the index to loc.
+  const start = sourceCode.getLocFromIndex(
+    comment.range[0] + '/*'.length + (match ? match.index + 1 : 0),
+  );
+  const end = {
+    line: start.line,
+    column: start.column + (match ? name.length : 1),
+  };
+
+  return { start, end };
+}

--- a/packages/eslint-plugin/typings/eslint-rules.d.ts
+++ b/packages/eslint-plugin/typings/eslint-rules.d.ts
@@ -830,17 +830,3 @@ declare module 'eslint/lib/rules/prefer-const' {
   >;
   export = rule;
 }
-
-declare module 'eslint/lib/rules/utils/ast-utils' {
-  import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
-
-  const utils: {
-    getNameLocationInGlobalDirectiveComment(
-      sourceCode: TSESLint.SourceCode,
-      comment: TSESTree.Comment,
-      name: string,
-    ): TSESTree.SourceLocation;
-  };
-
-  export = utils;
-}

